OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/ide/instances/lcp_on_fields/problem/LinearConstantPropagationProblemExtended.scala

```diff
@@ -0,0 +1,280 @@
+            .foldLeft(UnknownValue: LinearConstantPropagationValue) {
+                case (value, ArrayValue(initValue, values)) =>
+                    val arrayValue = index match {
+                        case Some(i) => values.getOrElse(i, initValue)
+                        case None =>
+                            if (values.values.forall { v => v == initValue }) {
+                                initValue
+                            } else {
+                                VariableValue
+                            }
+                    }
+                    lattice.meet(value, arrayValue)
+
+                case (_, LCPVariableValue) =>
+                    VariableValue
+            }
+    }
+
+    override def getNormalEdgeFunctionForGetField(
+        getFieldExpr: GetField[JavaStatement.V]
+    )(
+        source:     JavaStatement,
+        sourceFact: LinearConstantPropagationFact,
+        target:     JavaStatement,
+        targetFact: LinearConstantPropagationFact
+    )(implicit propertyStore: PropertyStore): EdgeFunctionResult[LinearConstantPropagationValue] = {
+        val objectVar = getFieldExpr.objRef.asVar
+        val fieldName = getFieldExpr.name
+
+        val lcpOnFieldsEOptionP =
+            propertyStore((source.method, source), LCPOnFieldsPropertyMetaInformation.key)
+
+        /* Decide based on the current result of the LCP on fields analysis */
+        lcpOnFieldsEOptionP match {
+            case FinalP(property) =>
+                val value = getObjectFieldFromProperty(objectVar, fieldName)(property)
+                FinalEdgeFunction(value match {
+                    case UnknownValue     => UnknownValueEdgeFunction
+                    case ConstantValue(c) => LinearCombinationEdgeFunction(0, c, lattice.top)
+                    case VariableValue    => VariableValueEdgeFunction
+                })
+
+            case InterimUBP(property) =>
+                val value = getObjectFieldFromProperty(objectVar, fieldName)(property)
+                value match {
+                    case UnknownValue =>
+                        InterimEdgeFunction(UnknownValueEdgeFunction, immutable.Set(lcpOnFieldsEOptionP))
+                    case ConstantValue(c) =>
+                        InterimEdgeFunction(
+                            LinearCombinationEdgeFunction(0, c, lattice.top),
+                            immutable.Set(lcpOnFieldsEOptionP)
+                        )
+                    case VariableValue =>
+                        FinalEdgeFunction(VariableValueEdgeFunction)
+                }
+
+            case _ =>
+                InterimEdgeFunction(UnknownValueEdgeFunction, immutable.Set(lcpOnFieldsEOptionP))
+        }
+    }
+
+    private def getObjectFieldFromProperty(
+        objectVar: JavaStatement.V,
+        fieldName: String
+    )(property: LCPOnFieldsPropertyMetaInformation.Self): LinearConstantPropagationValue = {
+        property
+            .results
+            .filter {
```

Comment:
Consider using `.collect` instead of `.filter{ ... }.map{ ... }` as it is more efficient. Alternatively, it might be possible to integrate the `map` (or even the `filter`) into the `foldLeft`. Also consider using `fold` instead of `foldLeft` if possible.